src/complex/DataStructure/DataStore.hpp

```diff
@@ -340,31 +338,31 @@ class DataStore : public AbstractDataStore<T>
     }
 
     // Write shape attributes to the dataset
-    auto tupleAttribute = datasetWriter.createAttribute(k_TupleShape);
+    auto tupleAttribute = datasetWriter.createAttribute(IDataStore::k_TupleShape);
     err = tupleAttribute.writeVector({m_TupleShape.size()}, m_TupleShape);
     if(err < 0)
     {
       return err;
     }
 
-    auto componentAttribute = datasetWriter.createAttribute(k_ComponentShape);
+    auto componentAttribute = datasetWriter.createAttribute(AbstractDataStore<T>::k_ComponentShape);
     err = componentAttribute.writeVector({m_ComponentShape.size()}, m_ComponentShape);
 
     return err;
   }
 
-  static std::unique_ptr<DataStore> readHdf5(const H5::DatasetReader& datasetReader)
+  static std::unique_ptr<AbstractDataStore<T>> ReadHdf5(const H5::DatasetReader& datasetReader)
```

Comment:
I think this function should still work in `DataArrayFactory.hpp` with `std::unique_ptr<DataStore>` as the return type. And `std::unique_ptr<EmptyStore>` for `EmptyStore` as well.

Reply:
I had to change the return type due to compile errors after collapsing an above request using these methods down to one line.

Reply:
```cpp
std::unique_ptr<AbstractDataStore<K>> dataStore = preflight ? std::unique_ptr<AbstractDataStore<K>>(EmptyDataStore<K>::readHdf5(datasetReader)) : DataStore<K>::readHdf5(datasetReader);
```
I think this would also fix that issue.